cider-dap/src/adapter.rs

```diff
@@ -131,37 +137,50 @@ impl MyAdapter {
         self.threads.clone()
     }
 
-    //Returns a dummy stack frame, set to change.
-    pub fn create_stack(&mut self) -> Vec<StackFrame> {
-        let frame = StackFrame {
-            id: self.stack_count.increment(),
-            // Maybe automate the name in the future?
-            name: String::from("Frame"),
-            source: Some(Source {
-                name: None,
-                path: Some(self.source.clone()),
-                source_reference: None,
-                presentation_hint: None,
-                origin: None,
-                sources: None,
-                adapter_data: None,
-                checksums: None,
-            }),
-            line: 1,
-            column: 0,
-            end_line: None,
-            end_column: None,
-            can_restart: None,
-            instruction_pointer_reference: None,
-            module_id: None,
-            presentation_hint: None,
-        };
-        self.stack_frames.push(frame);
-        // Return all stack frames
+    /// returns all frames in program
+    pub fn get_stack(&mut self) -> Vec<StackFrame> {
+        if self.stack_frames.is_empty() {
+            self.create_stack();
+        }
         self.stack_frames.clone()
     }
 
-    pub fn clone_stack(&self) -> Vec<StackFrame> {
+    /// creates call stack where each frame is a component
+    fn create_stack(&mut self) -> Vec<StackFrame> {
+        let components = self.debugger.get_components();
+        //turn the names into stack frames, ignore lines for right now
+        let frames = components.map(|(idx, comp)| {
+            {
```

Comment:
While I love a good iterator, I think this would be more clear as a `for` loop
which directly pushes the frame onto `stack_frames` rather than relying on the
iterator being consumed by extends